lib/acl.c

```diff
@@ -45,12 +45,31 @@ enum {
 
 /* Callback function for when there's no classification match. */
 static inline int
-drop_ipv6_acl_pkts(struct rte_mbuf **pkts, unsigned int num_pkts,
-	__attribute__((unused)) struct gatekeeper_if *iface)
+match_ipv6_acl_pkts(struct rte_mbuf **pkts, unsigned int num_pkts,
+	struct gatekeeper_if *iface)
 {
 	unsigned int i;
-	for (i = 0; i < num_pkts; i++)
-		rte_pktmbuf_free(pkts[i]);
+	for (i = 0; i < num_pkts; i++) {
+		unsigned int j;
+		bool free_pkt = false;
+		/* Start from the first one. */
+		for (j = 1; j < iface->acl_func_count; j++) {
+			int ret = iface->ext_funcs[j](pkts[i], iface);
+			if (ret == -ENOENT)
+				free_pkt = true;
+			else {
```

Comment:
For ``ret == 0``, it means that the packet has been successfully submitted, so there is no need to free the packet here, i.e., ``free_pkt = false``. For other errors, the ``submit_xxx()`` function will free the packet, so again we don't need to free it here.